packages/app/src/explorer/EntityListContainer.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import { useCallback, useRef } from 'react';
+
+import EntityList from './EntityList';
+import { getActiveButtonIndex } from './utils';
+
+interface Props {
+  selectedPath: string;
+  onSelect: (path: string) => void;
+}
+
+function EntityListContainer(props: Props) {
+  const { selectedPath, onSelect } = props;
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const focusOnNext = useCallback(() => {
+    const res = getActiveButtonIndex(ref.current);
+    if (!res) {
+      return;
+    }
+
+    const [i, buttonElementList] = res;
+    if (i < 0 || i === buttonElementList.length - 1) {
+      return;
+    }
+    buttonElementList[i + 1].focus();
+  }, []);
+
+  const focusOnPrevious = useCallback(() => {
+    const res = getActiveButtonIndex(ref.current);
```

Comment:
Returning an array with the index of the button and the list of buttons may not be the simplest abstraction. I feel like this could be simpler if you just had a function `getActiveButton` that would return the active element if it is an explorer button, else `undefined`. The rest can be inlined (listing all the buttons, finding the index of the active one, and focusing the next/previous one) unless a better abstraction is found.